torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -238,6 +321,48 @@ TORCH_CUDA_API Val* sum(Val* v1, const std::vector<int>& axes) {
   return reductionOp(BinaryOpType::Add, axes, init, v1);
 }
 
+TORCH_CUDA_API TensorView* broadcast(
+    TensorView* inp,
+    const std::vector<bool>& is_broadcast_dim) {
+  auto nBCastDims = is_broadcast_dim.size();
+  // Validate is_broadcast_dim
+  unsigned int n_broadcasts = 0;
+  for (auto ent : is_broadcast_dim)
+    if (ent)
+      n_broadcasts++;
+  TORCH_CHECK(
+      nBCastDims - n_broadcasts == inp->nDims(),
+      "Invalid broadcast, number of false entries in is_broadcast_dim expected to be ",
+      inp->nDims(),
+      " but received ",
+      nBCastDims - n_broadcasts);
+
+  if (n_broadcasts == 0) {
+    auto identity = unaryOp(UnaryOpType::Set, inp);
+    TORCH_INTERNAL_ASSERT(
+        identity->getValType().value() == ValType::TensorView,
+        "Expected identity op, but didn't get a TensorView back.");
+    return static_cast<TensorView*>(identity);
+  }
+
+  std::vector<IterDomain*> out_domain;
+  unsigned int iinp = 0, ibdim = 0;
+  while (ibdim < is_broadcast_dim.size()) {
+    if (is_broadcast_dim[ibdim]) {
+      out_domain.push_back(new IterDomain(
+          new Int(0), new Int(1), ParallelType::Serial, false, false, true));
```

Comment:
note to myself: broadcast is supported, I need to update integration to remove the conversion to implicit broadcast (stride-0-with-broadcasted-shape).

Reply:
Definitely not supported. Don't do that.